Quarkus projects do not compile when project path contains whitespaces

**Describe the bug**
I am trying to build a quarkus project on our company jenkins. The path to the project in jenkins contains whitespaces.
"/var/lib/jenkins/jobs/wolf - project/workspace/project-web/target/classes"
The build fails.
If I remove these whitespaces, the build is successfull.

**Expected behavior**
The project builds without errors caused by internal quarkus maven plugin problems.

**Actual behavior**
Build fails with the following error:
```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:0.18.0:build (default) on project documentrepository-web: Failed to build a runnable JAR: Failed to build a runner jar: Failed to augment application classes: Build failure: Build failed due to errors
[ERROR] [error]: Build step io.quarkus.deployment.index.ApplicationArchiveBuildStep#build threw an exception: java.lang.RuntimeException: java.io.FileNotFoundException: /var/lib/jenkins/jobs/wolf%20-%20project/workspace/project-web/target/classes (Datei oder Verzeichnis nicht gefunden)
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :project-web
```


**To Reproduce**
Steps to reproduce the behavior:
1. Move an existing, successfully compiling quarkus project into a path with whitespaces
2. Compile
3. Does not compile

**Additional context**
quarkus 018.0
java 8
Maven 3.6.1

@machi1990 
I tried to create a simple reproducer app, but I am unable to.
My project is a multi module project, with one module containing service classes, and the other contains the quarkus plugin.

I am however able to reproduce this error locally (on Windows 10) with the same app. I am running "mvn clean install -X"

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:0.18.0:build (default) on project documentrepository-web: Failed to build a runnable JAR: Failed to build a runner jar: Failed to augment application classes: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.deployment.index.ApplicationArchiveBuildStep#build threw an exception: java.lang.RuntimeException: java.io.FileNotFoundException: C:\workspaces\workspace\quarkus_repoducer_3034\folder%20with%20whitespaces\documentrepository\documentrepository-web\target\classes (Das System kann den angegebenen Pfad nicht finden)
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.quarkus:quarkus-maven-plugin:0.18.0:build (default) on project documentrepository-web: Failed to build a runnable JAR
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to build a runnable JAR
    at io.quarkus.maven.BuildMojo.execute (BuildMojo.java:211)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: io.quarkus.creator.AppCreatorException: Failed to build a runner jar
    at io.quarkus.creator.phase.runnerjar.RunnerJarPhase.provideOutcome (RunnerJarPhase.java:213)
    at io.quarkus.creator.phase.runnerjar.RunnerJarPhase.provideOutcome (RunnerJarPhase.java:59)
    at io.quarkus.creator.outcome.OutcomeResolver.resolve (OutcomeResolver.java:31)
    at io.quarkus.creator.AppCreator.resolveOutcome (AppCreator.java:239)
    at io.quarkus.maven.BuildMojo.execute (BuildMojo.java:203)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: io.quarkus.creator.AppCreatorException: Failed to augment application classes
    at io.quarkus.creator.phase.augment.AugmentPhase.doProcess (AugmentPhase.java:405)
    at io.quarkus.creator.phase.augment.AugmentPhase.provideOutcome (AugmentPhase.java:231)
    at io.quarkus.creator.phase.augment.AugmentPhase.provideOutcome (AugmentPhase.java:71)
    at io.quarkus.creator.outcome.OutcomeResolver.resolve (OutcomeResolver.java:31)
    at io.quarkus.creator.AppCreator.resolveOutcome (AppCreator.java:239)
    at io.quarkus.creator.phase.runnerjar.RunnerJarPhase.provideOutcome (RunnerJarPhase.java:211)
    at io.quarkus.creator.phase.runnerjar.RunnerJarPhase.provideOutcome (RunnerJarPhase.java:59)
    at io.quarkus.creator.outcome.OutcomeResolver.resolve (OutcomeResolver.java:31)
    at io.quarkus.creator.AppCreator.resolveOutcome (AppCreator.java:239)
    at io.quarkus.maven.BuildMojo.execute (BuildMojo.java:203)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.deployment.index.ApplicationArchiveBuildStep#build threw an exception: java.lang.RuntimeException: java.io.FileNotFoundException: C:\workspaces\workspace\quarkus_repoducer_3034\folder%20with%20whitespaces\documentrepository\documentrepository-web\target\classes (Das System kann den angegebenen Pfad nicht finden)
    at io.quarkus.builder.Execution.run (Execution.java:108)
    at io.quarkus.builder.BuildExecutionBuilder.execute (BuildExecutionBuilder.java:121)
    at io.quarkus.deployment.QuarkusAugmentor.run (QuarkusAugmentor.java:104)
    at io.quarkus.creator.phase.augment.AugmentPhase.doProcess (AugmentPhase.java:325)
    at io.quarkus.creator.phase.augment.AugmentPhase.provideOutcome (AugmentPhase.java:231)
    at io.quarkus.creator.phase.augment.AugmentPhase.provideOutcome (AugmentPhase.java:71)
    at io.quarkus.creator.outcome.OutcomeResolver.resolve (OutcomeResolver.java:31)
    at io.quarkus.creator.AppCreator.resolveOutcome (AppCreator.java:239)
    at io.quarkus.creator.phase.runnerjar.RunnerJarPhase.provideOutcome (RunnerJarPhase.java:211)
    at io.quarkus.creator.phase.runnerjar.RunnerJarPhase.provideOutcome (RunnerJarPhase.java:59)
    at io.quarkus.creator.outcome.OutcomeResolver.resolve (OutcomeResolver.java:31)
    at io.quarkus.creator.AppCreator.resolveOutcome (AppCreator.java:239)
    at io.quarkus.maven.BuildMojo.execute (BuildMojo.java:203)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.RuntimeException: java.io.FileNotFoundException: C:\workspaces\workspace\quarkus_repoducer_3034\folder%20with%20whitespaces\documentrepository\documentrepository-web\target\classes (Das System kann den angegebenen Pfad nicht finden)
    at io.quarkus.deployment.index.ApplicationArchiveBuildStep$1.apply (ApplicationArchiveBuildStep.java:243)
    at io.quarkus.deployment.index.ApplicationArchiveBuildStep$1.apply (ApplicationArchiveBuildStep.java:221)
    at java.util.HashMap.computeIfAbsent (HashMap.java:1127)
    at io.quarkus.deployment.index.ApplicationArchiveBuildStep.handleJarPath (ApplicationArchiveBuildStep.java:221)
    at io.quarkus.deployment.index.ApplicationArchiveBuildStep.indexPaths (ApplicationArchiveBuildStep.java:145)
    at io.quarkus.deployment.index.ApplicationArchiveBuildStep.scanForOtherIndexes (ApplicationArchiveBuildStep.java:110)
    at io.quarkus.deployment.index.ApplicationArchiveBuildStep.build (ApplicationArchiveBuildStep.java:85)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at io.quarkus.deployment.ExtensionLoader$1.execute (ExtensionLoader.java:507)
    at io.quarkus.builder.BuildContext.run (BuildContext.java:398)
    at org.jboss.threads.ContextClassLoaderSavingRunnable.run (ContextClassLoaderSavingRunnable.java:35)
    at org.jboss.threads.EnhancedQueueExecutor.safeRun (EnhancedQueueExecutor.java:2011)
    at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask (EnhancedQueueExecutor.java:1538)
    at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run (EnhancedQueueExecutor.java:1429)
    at java.lang.Thread.run (Thread.java:748)
    at org.jboss.threads.JBossThread.run (JBossThread.java:479)
Caused by: java.io.FileNotFoundException: C:\workspaces\workspace\quarkus_repoducer_3034\folder%20with%20whitespaces\documentrepository\documentrepository-web\target\classes (Das System kann den angegebenen Pfad nicht finden)
    at java.util.zip.ZipFile.open (Native Method)
    at java.util.zip.ZipFile.<init> (ZipFile.java:225)
    at java.util.zip.ZipFile.<init> (ZipFile.java:155)
    at java.util.jar.JarFile.<init> (JarFile.java:166)
    at java.util.jar.JarFile.<init> (JarFile.java:130)
    at io.quarkus.deployment.index.ApplicationArchiveBuildStep$1.apply (ApplicationArchiveBuildStep.java:225)
    at io.quarkus.deployment.index.ApplicationArchiveBuildStep$1.apply (ApplicationArchiveBuildStep.java:221)
    at java.util.HashMap.computeIfAbsent (HashMap.java:1127)
    at io.quarkus.deployment.index.ApplicationArchiveBuildStep.handleJarPath (ApplicationArchiveBuildStep.java:221)
    at io.quarkus.deployment.index.ApplicationArchiveBuildStep.indexPaths (ApplicationArchiveBuildStep.java:145)
    at io.quarkus.deployment.index.ApplicationArchiveBuildStep.scanForOtherIndexes (ApplicationArchiveBuildStep.java:110)
    at io.quarkus.deployment.index.ApplicationArchiveBuildStep.build (ApplicationArchiveBuildStep.java:85)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at io.quarkus.deployment.ExtensionLoader$1.execute (ExtensionLoader.java:507)
    at io.quarkus.builder.BuildContext.run (BuildContext.java:398)
    at org.jboss.threads.ContextClassLoaderSavingRunnable.run (ContextClassLoaderSavingRunnable.java:35)
    at org.jboss.threads.EnhancedQueueExecutor.safeRun (EnhancedQueueExecutor.java:2011)
    at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask (EnhancedQueueExecutor.java:1538)
    at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run (EnhancedQueueExecutor.java:1429)
    at java.lang.Thread.run (Thread.java:748)
    at org.jboss.threads.JBossThread.run (JBossThread.java:479)
[ERROR]
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :documentrepository-web
PS C:\workspaces\workspace\quarkus_repoducer_3034\folder with whitespaces\documentrepository>
```

--- a/core/deployment/src/main/java/io/quarkus/deployment/index/ApplicationArchiveBuildStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/index/ApplicationArchiveBuildStep.java
@@ -1,5 +1,6 @@
 package io.quarkus.deployment.index;
 
+import java.io.File;
 import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStream;
@@ -171,13 +172,13 @@ public class ApplicationArchiveBuildStep {
                 String jarPath = url.getPath().substring(0, url.getPath().lastIndexOf('!'));
                 return Paths.get(new URI(jarPath));
             } else if (url.getProtocol().equals("file")) {
-                int index = url.getPath().lastIndexOf("/META-INF");
+                final Path path = Paths.get(url.toURI());
+                final String filePath = path.toString();
+                final int index = filePath.lastIndexOf(File.separator + "META-INF");
                 if (index == -1) {
-                    return Paths.get(url.getPath());
+                    return path;
                 }
-                String pathString = url.getPath().substring(0, index);
-                Path path = Paths.get(new URI(url.getProtocol(), url.getHost(), pathString, null));
-                return path;
+                return Paths.get(filePath.substring(0, index));
             }
             throw new RuntimeException("Unknown URL type " + url.getProtocol());
         } catch (URISyntaxException e) {

--- a/core/deployment/src/test/java/io/quarkus/deployment/index/ApplicationArchiveBuildStepTestCase.java
+++ b/core/deployment/src/test/java/io/quarkus/deployment/index/ApplicationArchiveBuildStepTestCase.java
@@ -13,8 +13,13 @@ public class ApplicationArchiveBuildStepTestCase {
     @Test
     public void testUrlToPath() throws MalformedURLException {
         assertEquals(Paths.get("/a/path"), urlToPath(new URL("jar:file:/a/path!/META-INF/services/my.Service")));
+        assertEquals(Paths.get("/a/path with whitespace"),
+                urlToPath(new URL("jar:file:/a/path%20with%20whitespace!/META-INF/services/my.Service")));
         assertEquals(Paths.get("/a/path"), urlToPath(new URL("file:/a/path/META-INF/services/my.Service")));
+        assertEquals(Paths.get("/a/path with whitespace"),
+                urlToPath(new URL("file:/a/path%20with%20whitespace/META-INF/services/my.Service")));
         assertEquals(Paths.get("/a/path"), urlToPath(new URL("file:/a/path")));
+        assertEquals(Paths.get("/a/path with whitespace"), urlToPath(new URL("file:/a/path%20with%20whitespace")));
     }
 
     @Test(expected = RuntimeException.class)

